Resources/ProtobufDumper/ProtobufDumper/ProtobufCollector.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using google.protobuf;
+using ProtoBuf;
+
+namespace ProtobufDumper
+{
+    class ProtobufCollector
+    {
+        public List<FileDescriptorProto> Candidates { get; private set; }
+
+        public enum CandidateResult
+        {
+            OK,
+            Rescan,
+            Invalid
+        }
+
+        public ProtobufCollector()
+        {
+            Candidates = new List<FileDescriptorProto>();
+        }
+
+        public bool TryParseCandidate( string name, ReadOnlySpan<byte> data, out CandidateResult result, out Exception error )
+        {
+            FileDescriptorProto candidate;
+
+            try
+            {
+                using ( var ms = new MemoryStream( data.Length ) )
```

Comment:
Could use ArrayPool here rather than letting MemoryStream allocate (and regrow?) it’s own buffer.

Reply:
As in, I should allocate a buffer for MemoryStream and copy the Span into the buffer via MemoryStream?